core/src/main/java/com/linecorp/armeria/client/limit/ConcurrencyLimit.java

```diff
@@ -54,6 +56,52 @@ static ConcurrencyLimitBuilder builder(int maxConcurrency) {
         return new ConcurrencyLimitBuilder(maxConcurrency == MAX_VALUE ? 0 : maxConcurrency);
     }
 
+    /**
+     * Returns a newly-created {@link ConcurrencyLimit} with the specified {@code maxConcurrency}. For example:
+     * <pre> {@code
+     *     ConcurrencyLimit.builder(new DynamicLimit());
+     *
+     *     class DynamicLimit extends CachedValue<Integer> {
+     *         public DynamicLimit() {
+     *             super(defaultValue);
+     *             dogma.watcher().watch((rev, value) -> {
+     *                 set(Integer.parse(value));
+     *             });
+     *         }
+     *     }
+     * } </pre>
```

Comment:
I have some suggestions:
- This uses `ConcurrencyLimit.builder(...)`, but it's about `of(...)`. 😄 
- Users might not know what `dogma` is. Let's use a more expressional example.
- How about also using composition rather than inheritance? So we don't expose the `set` method to the outside and we can segregate the caching logic.

So we might do:
```java
/**
 * Returns a newly-created {@link ConcurrencyLimit} with the specified {@link Supplier}.
 * {@link Supplier#get()} might be frequently called, so please consider using {@link CachingSupplier} if
 * supplying the value needs a heavy computation. For example:
 *
 * <pre>{@code
 * ConcurrencyLimit limit = ConcurrencyLimit.of(new DynamicLimit());
 *
 * class DynamicLimit extends Supplier<Integer> {
 *
 *     private final CachingSupplier<Integer> cachingSupplier = CachingSupplier.of(3);
 *
 *     DynamicLimit() {
 *         LimitChangeListener listener = ...
 *         listener.addListener(updatedValue -> cachingSupplier.set(updatedValue));
 *     }
 *
 *     public final T get() {
 *         return limit.get();
 *     }
 * }
```

Reply:
I have modified the Javadoc and the example since I left this comment. PTAL. 🙇‍♂️